src/TeamStore.Keeper/Services/AssetService.cs

```diff
@@ -193,6 +194,39 @@ IApplicationIdentityService applicationIdentityService
             return retrievedAssets;
         }
 
+        // searching assets should preload all assets
+        public async Task<List<Asset>> GetAssetResultsAsync(string searchToken)
+        {
+            // Validate current user
+            var currentUser = await _applicationIdentityService.GetCurrentUser();
+            if (currentUser == null) throw new Exception("Unauthorised requests are not allowed."); ;
+
+            // Get assets with appropriate access checks
+            var retrievedAssets = await _dbContext.Assets.Where(a =>
+                a.IsArchived == false &&
+                a.Project.AccessIdentifiers.Any(ai => ai.Identity.Id == currentUser.Id))
+                .ToListAsync();
```

Comment:
This is the query that needs fixing. You need .Title "Is not null or whitespace" then .Title.StartsWith the search prefix so it is part of the SQL query.

If that works, do a .Title.ToLowerInvariant().StartsWith -> I know that .ToLower doesn't work well with SQLite EF.